libs/category/state/src/effects/category-page-filter.effects.spec.ts

```diff
@@ -26,158 +31,387 @@ import {
+          ...stubCategoryPageMetadata,
+          filter_requests: daffCategoryFiltersToRequests(stubCategoryPageMetadata.filters),
+        });
+      });
+    });
+
+    describe('and the driver call fails', () => {
+      beforeEach(() => {
+        driverGetSpy.and.returnValue(hot('#', {}, error));
+      });
+
+      it('should emit DaffCategoryPageLoadFailure with the transformed error', () => {
+        expected = cold('--a', { a: categoryPageLoadFailureAction });
+        expect(effects.updateFilters$()).toBeObservable(expected);
+        expect(daffCategoryDriver.get).toHaveBeenCalledWith({
+          ...stubCategoryPageMetadata,
+          filter_requests: daffCategoryFiltersToRequests(stubCategoryPageMetadata.filters),
+        });
+      });
+    });
+  });
+
+  describe('when CategoryPageReplaceFiltersAction is triggered', () => {
+    let expected;
+
+    beforeEach(() => {
+      const replaceCategoryFiltersAction = new DaffCategoryPageReplaceFilters([equalFilterRequest]);
+      actions$ = hot('--a', { a: replaceCategoryFiltersAction });
+    });
+
+    describe('and the driver call succeeds', () => {
+      beforeEach(() => {
+        driverGetSpy.and.returnValue(of(stubCategoryResponse));
+      });
+
+      it('should call get category with filter requests transformed from state', () => {
+        expected = cold('--(ab)', { a: productGridLoadSuccessAction, b: categoryLoadSuccessAction });
+        expect(effects.updateFilters$()).toBeObservable(expected);
+        expect(daffCategoryDriver.get).toHaveBeenCalledWith({
+          ...stubCategoryPageMetadata,
+          filter_requests: daffCategoryFiltersToRequests(stubCategoryPageMetadata.filters),
+        });
+      });
+    });
+
+    describe('and the driver call fails', () => {
+      beforeEach(() => {
+        driverGetSpy.and.returnValue(hot('#', {}, error));
+      });
 
-      expected = cold('--(ab)', { a: productGridLoadSuccessAction, b: categoryLoadSuccessAction });
-      expect(effects.changeCategoryFilters$).toBeObservable(expected);
-      expect(daffCategoryDriver.get).toHaveBeenCalledWith({
-        ...stubCategoryPageConfigurationState,
-        filter_requests: [{
-          name: 'name',
-          type: DaffCategoryFilterType.Equal,
-          value: ['value'],
-        }],
+      it('should emit DaffCategoryPageLoadFailure with the transformed error', () => {
+        expected = cold('--a', { a: categoryPageLoadFailureAction });
+        expect(effects.updateFilters$()).toBeObservable(expected);
+        expect(daffCategoryDriver.get).toHaveBeenCalledWith({
+          ...stubCategoryPageMetadata,
+          filter_requests: daffCategoryFiltersToRequests(stubCategoryPageMetadata.filters),
+        });
       });
     });
   });
 
-  describe('when ToggleCategoryFilterAction is triggered', () => {
+  describe('when CategoryPageApplyFiltersAction is triggered', () => {
+    let expected;
+
+    beforeEach(() => {
```

Comment:
These are all the same test, can they be merged into a single function and then called for each action type?

Reply:
made it into 2 functions and called each